substrate/client/authority-discovery/src/worker/addr_cache.rs

```diff
@@ -172,6 +172,140 @@ fn addresses_to_peer_ids(addresses: &HashSet<Multiaddr>) -> HashSet<PeerId> {
 	addresses.iter().filter_map(peer_id_from_multiaddr).collect::<HashSet<_>>()
 }
 
+trait Codable: Encode + Decode {}
```

Comment:
I would like to add this to https://github.com/paritytech/parity-scale-codec repo, and adding support for `HashSet`/`HashMap` using this, would anyone mind?